### PR TITLE
saga

### LINKED ISSUE #18: saga

### PR DESCRIPTION
**Pull request by eyakauleva**
close #18

### CODE REVIEW

File: src/main/java/com/solvd/micro9/tickets/domain/command/SetTicketsUserIdToNullByUserIdCommand.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package com.solvd.micro9.tickets.domain.command;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+@Data
+@AllArgsConstructor
+public class SetTicketsUserIdToNullByUserIdCommand {
```
Comment (h1alexbel) [inline src/main/java/com/solvd/micro9/tickets/domain/command/SetTicketsUserIdToNullByUserIdCommand.java]: this name is too complex IMO

consider more elegant naming for you classes

File: src/main/java/com/solvd/micro9/tickets/messaging/KfConsumer.java [hunk 2]
```diff
@@ -12,15 +16,28 @@
...
+                    log.info("received value: {}", record.value());
+                    if (EsStatus.PENDING.equals(
+                            record.value().getStatus()
+                    )) {
+                        SetTicketsUserIdToNullByUserIdCommand command = new SetTicketsUserIdToNullByUserIdCommand(
+                                record.value().getEntityId()
+                        );
+                        commandHandler.apply(command);
+                    } else {
+                        ProcessTicketUpdateCommand command = new ProcessTicketUpdateCommand(
```
Comment (h1alexbel) [inline src/main/java/com/solvd/micro9/tickets/messaging/KfConsumer.java]: line is too long

File: src/main/java/com/solvd/micro9/tickets/messaging/KfConsumer.java [hunk 3]
```diff
@@ -12,15 +16,28 @@
...
     public void fetch() {
         receiver.receive()
                 .subscribe(record -> {
-                    log.info("received message: {}", record);
-                    ticketService.updateDeletedUserTickets(record.value());
+                    log.info("received value: {}", record.value());
+                    if (EsStatus.PENDING.equals(
+                            record.value().getStatus()
+                    )) {
+                        SetTicketsUserIdToNullByUserIdCommand command = new SetTicketsUserIdToNullByUserIdCommand(
```
Comment (h1alexbel) [inline src/main/java/com/solvd/micro9/tickets/messaging/KfConsumer.java]: line is too long, split it, please

File: src/main/java/com/solvd/micro9/tickets/service/impl/EsTicketCommandHandlerImpl.java [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+                            .payload(String.format(template, command.getUserId()))
+                            .status(EsStatus.SUBMITTED)
+                            .build();
+                    esTicketList.add(event);
+                })
+                .doOnComplete(() -> isStreamCompleted[0] = true)
+                .subscribeOn(Schedulers.boundedElastic())
+                .subscribe();
+
+        while (!isStreamCompleted[0]) {
```
Comment (h1alexbel) [inline src/main/java/com/solvd/micro9/tickets/service/impl/EsTicketCommandHandlerImpl.java:108]: @eyakauleva check #20

File: src/main/java/com/solvd/micro9/tickets/service/impl/EsTicketCommandHandlerImpl.java [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+                            .entityId(esTicket.getEntityId())
+                            .payload(String.format(template, command.getUserId()))
+                            .status(EsStatus.SUBMITTED)
+                            .build();
+                    esTicketList.add(event);
+                })
+                .doOnComplete(() -> isStreamCompleted[0] = true)
+                .subscribeOn(Schedulers.boundedElastic())
+                .subscribe();
+
```
Comment (h1alexbel) [inline src/main/java/com/solvd/micro9/tickets/service/impl/EsTicketCommandHandlerImpl.java]: remove empty lines

File: src/main/java/com/solvd/micro9/tickets/service/impl/EsTicketCommandHandlerImpl.java [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+        while (!isStreamCompleted[0]) {
+        } //TODO is there a better way to do?
+
```
Comment (h1alexbel) [inline src/main/java/com/solvd/micro9/tickets/service/impl/EsTicketCommandHandlerImpl.java]: remove empty lines

File: src/main/java/com/solvd/micro9/tickets/service/impl/EsTicketCommandHandlerImpl.java [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+                                    command.getUserId()
+                            );
+                            cache.put(RedisConfig.CACHE_KEY, event.getId(), ticketCache)
+                                    .subscribeOn(Schedulers.boundedElastic())
+                                    .subscribe();
+                            producer.send("New event", event);
+                        })
+                )
+                .subscribeOn(Schedulers.boundedElastic())
+                .subscribe();
```
Comment (h1alexbel) [inline src/main/java/com/solvd/micro9/tickets/service/impl/EsTicketCommandHandlerImpl.java:124]: you subscribes in this mehtod twicely, that's not good practice
[first](
https://github.com/eyakauleva/ticket-microservice/pull/19/files#diff-d8d3b07fee98e5459b29350f259b5ca997b336a07aeb16e3c0fdab0befaa1e29R107) and now

**Review by h1alexbel — CHANGES_REQUESTED** [review]

File: src/main/java/com/solvd/micro9/tickets/domain/command/SetTicketsUserIdToNullByUserIdCommand.java (hunk 1 above)
Comment (eyakauleva) [reply src/main/java/com/solvd/micro9/tickets/domain/command/SetTicketsUserIdToNullByUserIdCommand.java]: fixed 9099b38

File: src/main/java/com/solvd/micro9/tickets/messaging/KfConsumer.java (hunk 2 above)
Comment (eyakauleva) [reply src/main/java/com/solvd/micro9/tickets/messaging/KfConsumer.java]: fixed 9099b38

File: src/main/java/com/solvd/micro9/tickets/messaging/KfConsumer.java (hunk 3 above)
Comment (eyakauleva) [reply src/main/java/com/solvd/micro9/tickets/messaging/KfConsumer.java]: fixed 9099b38

File: src/main/java/com/solvd/micro9/tickets/service/impl/EsTicketCommandHandlerImpl.java (hunk 5 above)
Comment (eyakauleva) [reply src/main/java/com/solvd/micro9/tickets/service/impl/EsTicketCommandHandlerImpl.java]: fixed 9099b38

File: src/main/java/com/solvd/micro9/tickets/service/impl/EsTicketCommandHandlerImpl.java (hunk 6 above)
Comment (eyakauleva) [reply src/main/java/com/solvd/micro9/tickets/service/impl/EsTicketCommandHandlerImpl.java]: fixed 9099b38

**Review by h1alexbel — APPROVED** [review]

**Pull request merged**